tests/test_models.py

```diff
@@ -395,6 +395,35 @@ def test_mars_local_tms():
     assert bbox == syrtis_tms.bbox
 
 
+def test_mars_tms_construction():
+    mars_sphere_crs = pyproj.CRS.from_user_input("IAU_2015:49900")
+    extent = [-90, -180, 90, 180]
+    mars_tms = morecantile.TileMatrixSet.custom(
+        extent, mars_sphere_crs, id="MarsGeographicCRS"
+    )
```

Comment:
interestingly I don't think the issue is directly with the CRS but more with morecantile directly

Reply:
well because I think you should use `matrix_scale=[1, 2]` for this TMS

https://github.com/developmentseed/morecantile/blob/3fa4e2ee130fda34f811dcc044585d8b3a21df04/morecantile/models.py#L671-L674

Reply:
ahh I forgot completely to add that, now it looks like maybe just the bbox coordinates should be swapped? or is it expected that for inverted CRSs that they would behave this way?
```
mars_tms.bbox
BoundingBox(left=-90.0, bottom=-180.0, right=90.0, top=180.0)
mars_tms.xy_bbox
BoundingBox(left=-90.0, bottom=-180.0, right=90.0, top=180.0)
```

Reply:
yeah I think it should, and then also the matrix_scale to `[2, 1]`

Reply:
```python
mars_sphere_crs = pyproj.CRS.from_user_input("IAU_2015:49900")
extent = [-180, 90, 180, 90]
mars_tms = morecantile.TileMatrixSet.custom(
    extent, mars_sphere_crs, id="MarsGeographicCRS", matrix_scale=[2, 1])


mars_tms.xy_bbox
>> BoundingBox(left=-180.0, bottom=-90.0, right=180.0, top=90.0)

mars_tms.bbox
>> BoundingBox(left=-180.0, bottom=-90.0, right=180.0, top=90.0)

mars_tms.xy_bounds(0,0,3)
>> BoundingBox(left=-180.0, bottom=67.5, right=-157.5, top=90.0)
```